libs/sdk-ui-tests-e2e/cypress/tools/chart.ts

```diff
@@ -131,4 +133,24 @@ export class Chart {
             this.getElement().find(DATA_LABELS_SELECTOR).eq(i).contains(labels[i]).should("exist");
         }
     }
+
+    public assertShortenMetricNameInLegend(width: number) {
+        this.assertShortenMetricName(LEGEND_NAME_SELECTOR, width);
+    }
+
+    public hoverOnChart1stPointInHigh() {
+        cy.get(HIGHCHART_POINT_SELECTOR).first().as("firstElement");
+        cy.get("@firstElement").trigger("mouseover", { force: true });
+        return this;
+    }
+
+    public assertShortenMetricNameInTooltip(width: number) {
```

Comment:
verify metric name in legend is an edge case, this function is no need, only assertShortenMetricName is enough, when you want to verify shorten metric in legend, just pass the css directly to the spec file

Reply:
I updated